### PR TITLE
ci: add GH action workflow for releasing to GH

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR adds a GH action to automatically release to GH following conventional commits.

**Related issue(s)**
https://github.com/asyncapi/parser-api/pull/94#issuecomment-1519849250

cc @derberg @14richa

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14
+      - name: Add plugin for conventional commits
+        run: npm install conventional-changelog-conventionalcommits
```
Comment (derberg) [inline .github/workflows/release.yml:19]:
```suggestion
        run: npm install conventional-changelog-conventionalcommits@5.0.0
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        working-directory: ./.github/workflows
+      - name: Release to GitHub
+        working-directory: ./.github/workflows
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          GIT_AUTHOR_NAME: asyncapi-bot
+          GIT_AUTHOR_EMAIL: info@asyncapi.io
+          GIT_COMMITTER_NAME: asyncapi-bot
+          GIT_COMMITTER_EMAIL: info@asyncapi.io
+        run: npx semantic-release
```
Comment (derberg) [inline .github/workflows/release.yml:29]:
```suggestion
        run: npx semantic-release@19.0.4
```

**Review by derberg — COMMENTED** [review]
add releaserc

```
---
branches:
- master
# by default release workflow reacts on push not only to master. 
#This is why out of the box sematic release is configured for all these branches
- name: next-spec
  prerelease: true
- name: next-major
  prerelease: true
- name: next-major-spec
  prerelease: true
- name: beta
  prerelease: true
- name: alpha
  prerelease: true
plugins:
- - "@semantic-release/commit-analyzer"
  - preset: conventionalcommits
- - "@semantic-release/release-notes-generator"
  - preset: conventionalcommits
- "@semantic-release/github"
```

and I added 2 comments to make sure you use specific versions of packages and not get surprised by some future releases

**Comment by smoya** [comment]
Done + accepted suggestions.

> add releaserc
> 
> ```
> ---
> branches:
> - master
> # by default release workflow reacts on push not only to master. 
> #This is why out of the box sematic release is configured for all these branches
> - name: next-spec
>   prerelease: true
> - name: next-major
>   prerelease: true
> - name: next-major-spec
>   prerelease: true
> - name: beta
>   prerelease: true
> - name: alpha
>   prerelease: true
> plugins:
> - - "@semantic-release/commit-analyzer"
>   - preset: conventionalcommits
> - - "@semantic-release/release-notes-generator"
>   - preset: conventionalcommits
> - "@semantic-release/github"
> ```
> 
> and I added 2 comments to make sure you use specific versions of packages and not get surprised by some future releases

Done + accepted suggestions.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+    branches:
+      - master
+jobs: 
+  release:
+    name: 'Release to GitHub'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
```
Comment (derberg) [inline .github/workflows/release.yml:13]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
```
Comment (derberg) [inline .github/workflows/release.yml:15]:
```suggestion
        uses: actions/setup-node@v3
```

**Review by derberg — APPROVED** [review]
LGTM, not a codeowner though

**Review by jonaslagoni — APPROVED** [review]

**Pull request merged**